backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/cfg/ElasticsearchIndexSettings.java

```diff
@@ -8,16 +8,41 @@
 
 /**
  * Configuration properties for Elasticsearch indexes.
+ * <p>
+ * Constants in this class are to be appended to a prefix to form a property key;
+ * see {@link org.hibernate.search.engine.cfg.IndexSettings} for details.
  */
 public final class ElasticsearchIndexSettings {
 
 	private ElasticsearchIndexSettings() {
 	}
 
+	/**
+	 * The index lifecycle strategy to use, i.e. how to automatically administrate the index on startup/shutdown.
+	 * <p>
+	 * Expects an {@link ElasticsearchIndexLifecycleStrategyName} value, or a String representation of such value.
+	 * <p>
+	 * Defaults to {@link Defaults#LIFECYCLE_STRATEGY}.
+	 */
 	public static final String LIFECYCLE_STRATEGY = "lifecycle.strategy";
 
+	/**
+	 * The minimal required status of the index on startup, before Hibernate Search can start using it.
+	 * <p>
+	 * Expects an {@link ElasticsearchIndexStatus} value, or a String representation of such value.
+	 * <p>
+	 * Defaults to {@link Defaults#LIFECYCLE_REQUIRED_STATUS}.
+	 */
 	public static final String LIFECYCLE_REQUIRED_STATUS = "lifecycle.required_status";
```

Comment:
In all the doc, you're talking about minimal required status. And it resonates with the discussion we had the other day.

I would change this one to `LIFECYCLE_MINIMAL_REQUIRED_STATUS` and the one below to `LIFECYCLE_MINIMAL_REQUIRED_STATUS_WAIT_TIMEOUT`